File: baekjoon_problems/11047.py
n, k = map(int, input().split())
coins = []
need = 0
for i in range(n):
    coins.append(int(input()))

for i in reversed(coins):
    temp = int(k/i)
    if temp > 0:
        k -= (temp*i)
        need += temp
    
print(need)

File: baekjoon_problems_kotlin/11047.kt
fun main(args: Array<String>) {
    val br = System.`in`.bufferedReader()

    var (coin, price) = br.readLine()!!.split(" ").map {it.toInt()}
    var coinList = Array(coin, {0})
    var totalCoinCount = 0

    for (i in coinList.indices) {
        coinList[i] = br.readLine().toInt()
    }

    for(i in coinList.indices.reversed()) {

        if(coinList[i] <= price) {
            val temp = price % coinList[i]
            totalCoinCount += price / coinList[i]
            price = temp

            if(price == 0) break
        }
    }

    println(totalCoinCount)


}